services/src/pro/contexts/postgres.rs

```diff
@@ -362,62 +362,125 @@ mod tests {
-        )
-        .unwrap();
+    /// Setup database schema and return its name.
+    async fn setup_db() -> (tokio_postgres::Config, String) {
+        let mut db_config = get_config_element::<Postgres>().unwrap();
+        db_config.schema = format!("geoengine_test_{}", rand::thread_rng().next_u64()); // generate random temp schema
 
-        // TODO: clean schema before test
+        let mut pg_config = tokio_postgres::Config::new();
+        pg_config
+            .user(&db_config.user)
+            .password(&db_config.password)
+            .host(&db_config.host)
+            .dbname(&db_config.database);
 
-        let ctx = PostgresContext::new(config, tokio_postgres::NoTls)
+        // generate schema with prior connection
+        PostgresConnectionManager::new(pg_config.clone(), NoTls)
+            .connect()
+            .await
+            .unwrap()
+            .batch_execute(&format!("CREATE SCHEMA {};", &db_config.schema))
             .await
             .unwrap();
 
-        anonymous(&ctx).await;
+        // fix schema by providing `search_path` option
+        pg_config.options(&format!("-c search_path={}", db_config.schema));
 
-        let _user_id = user_reg_login(&ctx).await;
+        (pg_config, db_config.schema)
+    }
 
-        let session = ctx
-            .user_db()
-            .write()
+    /// Tear down database schema.
+    async fn tear_down_db(pg_config: tokio_postgres::Config, schema: &str) {
+        // generate schema with prior connection
+        PostgresConnectionManager::new(pg_config, NoTls)
+            .connect()
             .await
-            .login(UserCredentials {
-                email: "foo@bar.de".into(),
-                password: "secret123".into(),
-            })
+            .unwrap()
+            .batch_execute(&format!("DROP SCHEMA {} CASCADE;", schema))
             .await
             .unwrap();
+    }
+
+    async fn with_temp_context<F, Fut>(f: F)
+    where
+        F: FnOnce(PostgresContext<NoTls>) -> Fut + std::panic::UnwindSafe + Send + 'static,
+        Fut: Future<Output = ()> + Send,
+    {
+        let (pg_config, schema) = setup_db().await;
+
+        // catch all panics and clean up first…
+        let executed_fn = {
+            let pg_config = pg_config.clone();
+            std::panic::catch_unwind(move || {
+                tokio::task::block_in_place(move || {
+                    Handle::current().block_on(async move {
+                        let ctx = PostgresContext::new(pg_config, tokio_postgres::NoTls)
+                            .await
+                            .unwrap();
+                        f(ctx).await
+                    })
+                })
+            })
+        };
+
+        tear_down_db(pg_config, &schema).await;
+
+        // then throw errors afterwards
+        if let Err(err) = executed_fn {
+            std::panic::resume_unwind(err);
+        }
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
```

Comment:
is this really needed? the documentation says multi_threaded is the default flavor.

Reply:
yes, otherwise block_on doesn't work